src/js/libgeo/GeoOps.js

```diff
@@ -584,6 +584,43 @@ geoOps.OtherPointOnCircle.updatePosition = function(el) {
     el.homog = General.withUsage(pos, "Point");
 };
 
+
+// start and end assumed to be normalized at once!
+geoOps._helper.projectPointToSegmentCR = function(seg, pos, infseg) {
+    var line = seg.homog;
+    var tt = List.turnIntoCSList([line.value[0], line.value[1], CSNumber.zero]);
+    var farpoint = List.sub(seg.startpos, seg.endpos);
+    var cr = List.crossratio3(
+        farpoint, seg.startpos, seg.endpos, pos, tt);
+
+    //handle case if we have infinite segment
+    if (infseg) {
```

Comment:
After further investigations i came to the conclusion that this check is probably needed. When we create a segment we implicitly assume that it's finite and encoding far/midpoint accordingly. We can't assume that for arc simply because we don't know in which position the points occur when the arc degenerates. If you have a different opinion @gagern, i'm open to suggestions.